### PR TITLE
If both meta and doris-meta are present, Fe should exit to avoid starting with outdated metadata.

### PR DESCRIPTION
**Pull request by gengjun-git**
If `metaDir` is the default config: `StarRocksFE.STARROCKS_HOME_DIR + "/meta"`, we should check whether both the new default dir `(STARROCKS_HOME_DIR + "/meta")` and the old default dir `(DORIS_HOME_DIR + "/doris-meta")` are present. If both are present, we need to let users keep only one to avoid starting from outdated metadata.

### CODE REVIEW

**Review by imay — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java [hunk 1]
```diff
@@ -806,16 +807,38 @@ public void initialize(String[] args) throws Exception {
...
+        //      and the old default dir (DORIS_HOME_DIR + "/doris-meta") are present. If both are present,
+        //      we need to let users keep only one to avoid starting from outdated metadata.
+        String oldDefaultMetaDir = System.getenv("DORIS_HOME") + "/doris-meta";
+        String newDefaultMetaDir = StarRocksFE.STARROCKS_HOME_DIR + "/meta";
+        if (metaDir.equals(newDefaultMetaDir)) {
+            File oldMeta = new File(oldDefaultMetaDir);
+            File newMeta = new File(newDefaultMetaDir);
+            if (oldMeta.exists() && newMeta.exists()) {
+                LOG.error("New default meta dir: {} and Old default meta dir: {} are both present. " +
+                        "Please check which one has the latest data, and remove the another one.",
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java:821]: should let user use `meta` dir, not `doris-meta`

File: fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java [hunk 2]
```diff
@@ -806,16 +807,38 @@ public void initialize(String[] args) throws Exception {
...
+            }
+        }
         File meta = new File(metaDir);
         if (!meta.exists()) {
-            String oldMetaDir = System.getenv("DORIS_HOME") + "/doris-meta";
-            File oldMeta = new File(oldMetaDir);
+            // If metaDir is not the default config, it means the user has specified the other directory
+            // We should not use the oldDefaultMetaDir.
+            // Just exit in this case
+            if (!metaDir.equals(newDefaultMetaDir)) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java:831]: This logic is right?

Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java:831]: Why not? We cannot change the configuration that has been set by the user.

**Review by kangkaisen — APPROVED** [review]

**Review by imay — APPROVED** [review]
LGTM

**Pull request merged**